decorators_library/decorators.py

```diff
@@ -1 +1,89 @@
-# implement your decorators here.
+import signal
+from .exceptions import TimeoutError
+import logging
+import time
+
+def timeout_occured(signum, frame):
```

Comment:
interesting idea and implementation! will look at the documentation for `signal` when I have more time